src/main/java/com/pnu/dev/radioserviceapi/controller/admin/ScheduleAdminController.java

```diff
@@ -1,12 +1,202 @@
 package com.pnu.dev.radioserviceapi.controller.admin;
 
+import com.pnu.dev.radioserviceapi.dto.form.NewScheduleItemForm;
+import com.pnu.dev.radioserviceapi.dto.form.UpdateScheduleItemForm;
+import com.pnu.dev.radioserviceapi.dto.response.DayOfWeekResponse;
+import com.pnu.dev.radioserviceapi.dto.response.schedule.DailySchedule;
+import com.pnu.dev.radioserviceapi.dto.response.schedule.ScheduleItemDto;
+import com.pnu.dev.radioserviceapi.exception.RadioServiceAdminException;
+import com.pnu.dev.radioserviceapi.mongo.DayOfWeek;
+import com.pnu.dev.radioserviceapi.mongo.Program;
+import com.pnu.dev.radioserviceapi.service.ProgramService;
+import com.pnu.dev.radioserviceapi.service.ScheduleItemService;
+import com.pnu.dev.radioserviceapi.service.ScheduleService;
+import com.pnu.dev.radioserviceapi.util.HttpUtils;
+import com.pnu.dev.radioserviceapi.util.OperationResult;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static java.util.Objects.isNull;
 
 @Controller
 @RequestMapping("/admin/schedule")
 public class ScheduleAdminController {
 
-    // 1. update schedule for week (page with 7-day schedule and ability to update it)
-    // schedule item should contain time, program and optional description for single occurrence
+    private static final String FLASH_MESSAGE = "flashMessage";
+
+    private static final String FLASH_ERROR = "flashErrorMessage";
```

Comment:
Maybe it'll be better to create something like 
class FlashMessage { 
     enum type {ERROR, SUCCESS}
     String text
}

Reply:
We need to use it with the model map, where one of this consts will be the key and value will be whatever message you want to be displayed to the user, so I don't see the point of creating this dto.